src/content/resources/java/class.md

```diff
@@ -0,0 +1,141 @@
+
+Okay so we have added attributes to our animal which is great, but we want our animal to actually be able to *do* things. Let's pick 2
+things we want our animal to do.
+Our animal should
+* move
+* sleep
+
+These are verbs which means they will be implemented as methods.
+```java
+public class Animal {
+  private String name;
+  private int age;
+  private float weight;
+  
+  public void move() {
+    System.out.println("The animal moves.");
+  }
+  
+  public void sleep() {
+    System.out.println("The animal sleeps.");
+  }
+}
+```
+
+Now you can see what an animal will be like in our program. Each animal will have 3 attributes and 2 actions. There is one more thing
+that must be discussed in order to understand how classes work. Every class has to have a constructor. 
+
+## What is a constructor?
+A constructor is a method that enables a class to create actual objects. If we think of the class as the blueprint, the constructor is like the person using the blueprint to make the item.
+
+The constructor will always have the same name as the class and there will never be a return type.
+
+If you do not write a constructor for your class, it will be given one implicitely and will look like:
+```java
+public ClassName() {
+  super();
+}
+```
+The call to `super()` is calling the parameterless constructor of the parent class. If the parent class does not have a parameterless constructor, it will produce an error. 
+This constructor will give all of your class variables their default zero value (numbers are 0, boolean is false, strings and other objects are null). To instead assign your own values to them, you have to create a constructor that accepts parameters. 
+```java
+public ClassName(int someVar) {
+  this.someVar = someVar;
+}
+```
+Every constructor you write will call `super()` as the very first line it executes unless you call a different form of super such as `super(var1, var2)`.
+It is important to note that when you explicitly write a constructor in your class, the class will no longer create the parameterless constructor for you. So if you create a class with 1 constructor that accepts parameters like the one above, and then a child class implicitely calls `super()`, an error will occur. You would either need to supply a parameterless constructor in your parent class or make the child explicitly call the constructor with parameters.
+```java
+public ChildClassConstructor(int someVar) {
+  super(someVar);
+}
+```
+Applying this to our Animal class, lets create 2 constructors.
+1. a parameterless constructor that will implicitly call the parent constructor (via `super()`) and assign our variables to their zero value
+2. a constructor that accepts parameters, implicitely calls the parent constructor (via `super()`), and assign our variables with the parameter values
+
+We will also create a method for each variable that, when called, returns the value of that variable. These are called "getters".
```

Comment:
Are getters and setters essentials / basics when it comes to classes? Are they necessary to have in this example?

Reply:
I definitely did not intend to get into so many topics when writing about classes but there are so many things that need to be understood.
The getters are there so that I can show the difference between a default and custom object. Using a getter with private variables seemed to be the best course of action rather than making the variables public and risking that being a beginner's standard.
Thoughts?